tests_integration/test_integration.py

```diff
@@ -0,0 +1,298 @@
+
+    return scenario_id, scenario
+
+
+def is_proxy_running(port: int, timeout: float = 2.0) -> bool:
+    """
+    Returns if the proxy is running on localhost:port.
+
+    :param port: The port to check.
+    :param timeout: The timeout in seconds.
+
+    :return: A function that returns True if the proxy is running, False otherwise.
+    """
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+        sock.settimeout(timeout)
+        return sock.connect_ex(("localhost", port)) == 0
+
+
+@pytest.fixture
+def mitmproxy(xprocess: XProcess, scenario: tuple[int, dict[str, TestCase]]):
+    scenario_id, _ = scenario
+
+    class MitmproxyStarter(ProcessStarter):
+        args = ["./mock_proxy_start.sh"]
+        timeout = 60
+        popen_kwargs = {"cwd": TESTS_INTEGRATION_DIR}
+        env = os.environ | {
+            "SCENARIO_ID": str(scenario_id),
+            "PYTHONPATH": str(CF_SCRIPTS_ROOT_DIR.resolve()),
+        }
+
+        def startup_check(self):
+            return is_proxy_running(port=8080)
+
+    xprocess.ensure("mitmproxy", MitmproxyStarter)
+
+    yield
+
+    xprocess.getinfo("mitmproxy").terminate()
+
+
+@contextlib.contextmanager
+def in_fresh_cf_graph():
+    """
+    Context manager to execute code within the context with a new clone
+    of cf-graph in a temporary directory.
+    """
+    old_working_dir = os.getcwd()
+
+    with tempfile.TemporaryDirectory() as tmpdir_s:
+        tmpdir = Path(tmpdir_s)
+
+        cf_graph_repo = settings().graph_github_backend_repo
+
+        # --depth=5 is the same value as used in prod (see autotick-bot/install_bot_code.sh)
+        subprocess.run(
+            [
+                "git",
+                "clone",
+                "--depth=5",
+                f"https://github.com/{cf_graph_repo}.git",
+                "cf-graph",
+            ],
+            check=True,
+            cwd=tmpdir,
+        )
+
+        cf_graph_dir = tmpdir / "cf-graph"
+
+        subprocess.run(
+            [
+                "git",
+                "config",
+                "user.name",
+                "regro-cf-autotick-bot-staging",
+            ],
+            check=True,
+            cwd=cf_graph_dir,
+        )
+
+        subprocess.run(
+            [
+                "git",
+                "config",
+                "user.email",
+                "regro-cf-autotick-bot-staging@users.noreply.github.com",
+            ],
+            check=True,
+            cwd=cf_graph_dir,
+        )
+
+        subprocess.run(
+            [
+                "git",
+                "config",
+                "pull.rebase",
+                "false",
+            ],
+            check=True,
+            cwd=cf_graph_dir,
+        )
+
+        subprocess.run(
+            ["git", "config", 'http."https://github.com".proxy', '""'], check=True
+        )
+
+        os.chdir(cf_graph_dir)
+        yield
+
+    os.chdir(old_working_dir)
+
+
+@contextlib.contextmanager
+def mitmproxy_env():
```

Comment:
This can be done with `monkeypatch.setenv` if you want, but maybe this is more practical.

Reply:
refactoring to using `monkeypatch.setenv`: cd1007c22431d5b3b4e749f3a618b47ad60a7b9a

Reply:
You can use [`monkeypatch.context()`](https://docs.pytest.org/en/stable/reference/reference.html#pytest.MonkeyPatch.context)